test/unit/Ramm/getReserves.js

```diff
@@ -1,292 +1,236 @@
-    nxmA = eth.mul(nxmA).div(eth.sub(nrDenomAddend));
-  }
-
-  // apply ratchet below
-  // check if we should be using the ratchet or the book value price using:
-  // Nbv > Nr <=>
-  // ... <=>
-  // cap * n < e * sup + r * cap * n
-  const priceBufferB = PRICE_BUFFER_DENOMINATOR - PRICE_BUFFER;
-  const bufferedCapitalB = capital.mul(priceBufferB).div(PRICE_BUFFER_DENOMINATOR);
-
-  const kB = bufferedCapitalB.mul(nxmB);
-  const kBRatchetValue = nxmB
-    .mul(capital)
-    .mul(elapsed)
-    .mul(state.ratchetSpeed)
-    .div(RATCHET_PERIOD)
-    .div(RATCHET_DENOMINATOR);
-  const kBRatcheted = eth.mul(supply).add(kBRatchetValue);
-  if (kB.lt(kBRatcheted)) {
-    // use bv
-    nxmB = eth.mul(supply).div(bufferedCapitalB);
-  } else {
-    // use ratchet
-    const nrDenomAddend = nxmB
-      .mul(elapsed)
-      .mul(state.ratchetSpeed)
-      .mul(capital)
-      .div(supply)
-      .div(RATCHET_PERIOD)
-      .div(RATCHET_DENOMINATOR);
-    nxmB = eth.mul(nxmB).div(eth.add(nrDenomAddend));
-  }
-
-  return {
-    nxmA,
-    nxmB,
-    eth,
-    budget,
-    ratchetSpeed: state.ratchetSpeed,
-    timestamp: currentTimestamp,
-  };
-};
-
-/* ========== Set Slo1.ethReserve helper functions ========== */
+const FAST_RATCHET_SPEED = 5000;
+const INITIAL_LIQUIDITY = parseEther('5000');
+const INITIAL_BUDGET = parseEther('43835');
+
+const getRammState = paramState => ({
+  nxmA: INITIAL_LIQUIDITY.mul(parseEther('1')).div(SPOT_PRICE_A),
+  nxmB: INITIAL_LIQUIDITY.mul(parseEther('1')).div(SPOT_PRICE_B),
+  eth: INITIAL_LIQUIDITY,
+  budget: INITIAL_BUDGET,
+  ratchetSpeed: FAST_RATCHET_SPEED,
+  timestamp: Math.floor(Date.now() / 1000),
+  ...paramState,
+});
 
 /**
- * Sets the value of the Ether reserve in the RAMM contract.
+ * Calculates the expected ETH liquidity after extracting ETH
  *
- * @async
- * @param {string} rammAddress - The address of the RAMM contract
- * @param {number} valueInEther - The value of the Ether reserve in Ether
- * @return {Promise<void>}
+ * @param {Object} state - The current state object
+ * @param {number} nextBlockTimestamp - The timestamp of the next block
+ * @return {number} The expected amount of ETH to extract from the state
  */
-const setEthReserveValue = async (rammAddress, valueInEther) => {
-  const SLOT_1_POSITION = '0x3';
-  // Convert valueInEther to 128 bits wei hex value
-  const hexValueInWei = parseEther(valueInEther.toString()).toHexString();
-  const newEtherReserve = '0x' + removeHexPrefix(hexValueInWei).padStart(32, '0'); // 32 hex chars in 128 bits
-  // Get current Slot1 value
-  const slot1Value = await ethers.provider.send('eth_getStorageAt', [rammAddress, SLOT_1_POSITION]);
-  // Update Slot1 to have new ethReserve value
-  const newSlot1Value = await replaceHexValueInBitPos(slot1Value, newEtherReserve, 128);
-  return ethers.provider.send('hardhat_setStorageAt', [rammAddress, SLOT_1_POSITION, newSlot1Value]);
+const getExpectedEthExtract = (state, nextBlockTimestamp) => {
```

Comment:
In the context of the current function the timestamp is the timestamp when the tx happens. The fact that it's going to be the next block is abstracted away. I suggest renaming it to just `timestamp`.